llvm/test/CodeGen/DirectX/scalar-load.ll

```diff
@@ -0,0 +1,44 @@
+; RUN: opt -S -dxil-data-scalarization -scalarizer -scalarize-load-store -dxil-op-lower -mtriple=dxil-pc-shadermodel6.3-library %s | FileCheck %s
+; RUN: llc %s -mtriple=dxil-pc-shadermodel6.3-library --filetype=asm -o - | FileCheck %s
+@"arrayofVecData" = local_unnamed_addr addrspace(3) global [2 x <3 x float>] zeroinitializer, align 16
+@"vecData" = external addrspace(3) global <4 x i32>, align 4
+@staticArrayOfVecData = internal global [3 x <4 x i32>] [<4 x i32> <i32 1, i32 2, i32 3, i32 4>, <4 x i32> <i32 5, i32 6, i32 7, i32 8>, <4 x i32> <i32 9, i32 10, i32 11, i32 12>], align 4
```

Comment:
Could we have a test with multiple-dimension vector array?
Something like `[3 x [ 3 x <4 x i32>]]`

Reply:
I added that test and a load  to make sure GEPS are fine. also added a new test file with a 3 dimmensional array of vectors aswell. So the recursion seems to be working fine.